modules/apps/layout/layout-page-template-test/src/testIntegration/java/com/liferay/layout/page/template/internal/upgrade/v5_8_0/test/LayoutPageTemplateStructureRelUpgradeProcessTest.java

```diff
@@ -0,0 +1,259 @@
+import com.liferay.portal.kernel.service.LayoutLocalService;
+import com.liferay.portal.kernel.service.ServiceContextThreadLocal;
+import com.liferay.portal.kernel.service.change.tracking.CTService;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.version.Version;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+import com.liferay.segments.service.SegmentsExperienceLocalService;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Igor Costa
+ */
+@RunWith(Arquillian.class)
+public class LayoutPageTemplateStructureRelUpgradeProcessTest
+	extends BaseCTUpgradeProcessTestCase {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_objectDefinition = _objectDefinitionLocalService.addObjectDefinition(
+			RandomTestUtil.randomString(), TestPropsValues.getUserId(), 0, true,
+			false);
+
+		_objectDefinition.setName(RandomTestUtil.randomString());
+
+		_group = GroupTestUtil.addGroup();
+
+		_layout = LayoutTestUtil.addTypeContentLayout(_group);
+
+		_segmentsExperienceId =
+			_segmentsExperienceLocalService.fetchDefaultSegmentsExperienceId(
+				_layout.getPlid());
+
+		ServiceContextThreadLocal.pushServiceContext(
+			ServiceContextTestUtil.getServiceContext(
+				_group.getGroupId(), TestPropsValues.getUserId()));
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		_objectDefinitionLocalService.deleteObjectDefinition(_objectDefinition);
+
+		ServiceContextThreadLocal.popServiceContext();
+	}
+
+	@Test
+	public void testUpgrade() throws Exception {
+		String originalKey = StringBundler.concat(
+			"com.liferay.object.web.internal.info.collection.provider.",
+			"ObjectEntrySingleFormVariationInfoCollectionProvider_",
+			TestPropsValues.getCompanyId(), "_", _objectDefinition.getName());
+
+		String expectedKey = StringBundler.concat(
+			ObjectEntry.class.getName(), "#",
+			_objectDefinition.getObjectDefinitionId());
+
+		_objectDefinition.setClassName(expectedKey);
```

Comment:
Since I was add the `objectDefinition` by the following logic, it came with a empty className, therefore I needed to change it. But according to your suggestions, I don't need to add it like that no more.
```
_objectDefinitionLocalService.addObjectDefinition(
			RandomTestUtil.randomString(), TestPropsValues.getUserId(), 0, true,
			false);
```